src/utility/Albany_Accessors.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef ALBANY_ACCESSORS_HPP
+#define ALBANY_ACCESSORS_HPP
+
+#include "Albany_Meta_Utils.hpp"
+#include "PHAL_AlbanyTraits.hpp"
+
+
+namespace Albany
+{
+
+template<typename ST>
+struct Accessor {
+  ST& getValue () { return value; }
```

Comment:
I'm surprised we don't need a const method as well...

Edit: of course, since you only store shared_ptr<Accessor<...>>, it's always a non-const object. See comments in SharedParameter though.